### ISSUE TITLE
add pip install to automatic testing suite which checks for missing dependencies

### ISSUE THREAD

**Issue by iamtrask** [OpenMined/PySyft#5783]
Blocked by: https://github.com/OpenMined/PySyft/issues/5781

At the time of writing, the [hagrid tool](https://github.com/OpenMined/PySyft/tree/demo_strike_team_branch_4/packages/hagrid) requires a wide variety of dependencies in order to run its many commands. There are extra dependencies which are required for Linux because "docker compose" function is only available in beta (which must be specially installed).

In #5780, we created warnings for when a dependency is missing, and in #5781 we empowered the tool to install deps autoamtically. In this project, you're going to test this functionality and add such tests to CI so that we can ensure the functionality stays around (especially as versions change)

**Acceptance Criteria:**
- when merging a new PR - github's CI should automatically test to make sure that hagrid can be pip installed and executed (and that all dependencies are installed automatically) on OSX and Linux.

------

If you would like to claim this issue, you must first open a pull request with the following attributes:
- **Title:** [WIP]  add pip install to automatic testing suite which checks for missing dependencies
- **Branch merging into:** [iamtrask/ch65/pip-install-hagrid-and-it-s-deps-automagically](https://github.com/OpenMined/PySyft/tree/iamtrask/ch65/pip-install-hagrid-and-it-s-deps-automagically)
- **Description:** must include the text "[ch65]" and "#5777"

If you do not claim this issue by creating a pull request, others will assume you are not working on it and will pick up the task themselves (regardless of whether you comment on this issue).
